mypy/build.py

```diff
@@ -1250,8 +1265,17 @@ def write_cache(id: str, path: str, tree: MypyFile,
       corresponding to the metadata that was written (the latter may
       be None if the cache could not be written).
     """
+    # For Bazel we use relative paths and zero mtimes.
+    bazel = manager.options.bazel
+
     # Obtain file paths
-    path = os.path.abspath(path)
+    if not bazel:
+        # Normally, make all paths absolute.
+        path = os.path.abspath(path)
+    else:
+        # For Bazel, make all paths relative (else Bazel caching is thwarted).
+        # And also patch tree.path, which might have been made absolute earlier.
+        tree.path = path = os.path.relpath(path)
```

Comment:
Bazel uses a lot of symlinks, so you might want os.path.realpath here.
(Do you also want it for the "not bazel" case?)